### DISCUSSION TITLE
How can I make IntelliSense work for discord.js when using typescript?

### DISCUSSION THREAD

**Discussion by blnayan** [discordjs/discord.js#5876] [Q&A] [answered]
I am using typescript to make a discord bot but I ran into a problem which is that when using require function to import discord.js module, IntelliSense doesn't work. Is there a possible way to fix this?

If you want an example, then the code snippet at [discord.js.org](discord.js.org) homepage is what used similarly to import discord.js module.

**Comment by monbrey** [comment, marked as answer]
I would expect that if writing TypeScript you'd be using the ES6 `import` syntax rather than `require`.

**Reply by blnayan** [reply]
Is it possible to have IntelliSense while using `require` in TypeScript?

**Reply by advaith1** [reply]
You can use `import Discord = require('discord.js')` [[docs](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require)], but it's probably better to just use the standard `import Discord from 'discord.js'` syntax.

Not having IntelliSense with `const`/`let`/`var` and `import` seems to be standard TS behavior, not related to d.js.

**Reply by blnayan** [reply]
Yea I figured that out as I researched a bit more. Thanks though.
